src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,592 @@
+        """
+        Compute incidence angles from imaging list.
+
+        Column positions may differ between telescope types and sim_telarray builds.
+        Parse header lines ("#   Column N: ...") to discover indices; otherwise
+        fall back to legacy positions (1-based): focal=26, primary=32, secondary=36,
+        primary X/Y = 29/30, secondary X/Y = 33/34.
+        """
+        (
+            focal_idx,
+            primary_idx,
+            secondary_idx,
+            primary_x_idx,
+            primary_y_idx,
+            secondary_x_idx,
+            secondary_y_idx,
+        ) = self._discover_column_indices(photons_file)
+
+        col_idx = {
+            "focal": focal_idx,
+            "primary": primary_idx,
+            "secondary": secondary_idx,
+            "prim_x": primary_x_idx,
+            "prim_y": primary_y_idx,
+            "sec_x": secondary_x_idx,
+            "sec_y": secondary_y_idx,
+        }
+
+        focal = []
+        primary = [] if self.calculate_primary_secondary_angles else None
+        secondary = [] if self.calculate_primary_secondary_angles else None
+        # Primary-hit radius on M1 (computed from hit x/y in cm -> meters)
+        radius_m = [] if self.calculate_primary_secondary_angles else None
+        # Secondary-hit radius on M2 (computed from hit x/y in cm -> meters)
+        secondary_radius_m = [] if self.calculate_primary_secondary_angles else None
+        # Hit coordinates (meters), derived from cm columns
+        primary_hit_x_m = [] if self.calculate_primary_secondary_angles else None
+        primary_hit_y_m = [] if self.calculate_primary_secondary_angles else None
+        secondary_hit_x_m = [] if self.calculate_primary_secondary_angles else None
+        secondary_hit_y_m = [] if self.calculate_primary_secondary_angles else None
+
+        for parts in self._iter_data_rows(photons_file):
+            self._append_values(
+                parts,
+                col_idx,
+                focal,
+                primary,
+                secondary,
+                radius_m,
+                secondary_radius_m,
+                primary_hit_x_m,
+                primary_hit_y_m,
+                secondary_hit_x_m,
+                secondary_hit_y_m,
+            )
+
+        result = {"angle_incidence_focal_deg": focal}
+        if self.calculate_primary_secondary_angles:
+            result["angle_incidence_primary_deg"] = primary
+            result["angle_incidence_secondary_deg"] = secondary
+            result["primary_hit_radius_m"] = radius_m
+            result["secondary_hit_radius_m"] = secondary_radius_m
+            result["primary_hit_x_m"] = primary_hit_x_m
+            result["primary_hit_y_m"] = primary_hit_y_m
+            result["secondary_hit_x_m"] = secondary_hit_x_m
+            result["secondary_hit_y_m"] = secondary_hit_y_m
+        return result
+
+    def _discover_column_indices(self, photons_file):
```

Comment:
This is a bit hard to understand. What does 'discovered' mean? Found?

Ideal would be if there would be unit test called `test_discover_column_indices....` with some clear examples.

Reply:
Renamed and renamed the unit test that tests this method.